server/src/index.js

```diff
@@ -14,58 +14,67 @@ const io = socketio(server);
 app.use(cors());
 
 io.on("connection", (socket) => {
-  socket.on("join", ({ name, room, pfpSrc }, callback) => {
-    const { error, user } = addUser({ id: socket.id, name, room, pfpSrc });
+    socket.on("join", ({ name, room, pfpSrc }, callback) => {
+        const { error, user } = addUser({ id: socket.id, name, room, pfpSrc });
 
-    if (error) {
-      return callback(error);
-    }
+        if (error) {
+            return callback(error);
+        }
 
-    const roommates = getUsersInRoom(user.room);
+        const roommates = getUsersInRoom(user.room);
 
-    socket.join(user.room);
-    socket.emit("message", {
-      user: "admin",
-      text: `${user.name} welcome to the room ${user.room}`,
-      roommates,
+        socket.join(user.room);
+        socket.emit("message", {
+            user: "admin",
+            text: `${user.name} welcome to the room ${user.room}`,
+            roommates,
+        });
+
+        socket.broadcast.to(user.room).emit("message", {
+            user: "admin",
+            text: `${user.name} has joined!`,
+            roommates,
+        });
+        callback();
     });
 
-    socket.broadcast.to(user.room).emit("message", {
-      user: "admin",
-      text: `${user.name} has joined!`,
-      roommates,
+    socket.on("sendMessage", (message, callback) => {
+        const user = getUser(socket.id);
+        io.to(user.room).emit("message", {
+            user: user.name,
+            text: message,
+            pfpSrc: user.pfpSrc,
+        });
+        callback();
     });
-    callback();
-  });
 
-  socket.on("sendMessage", (message, callback) => {
-    const user = getUser(socket.id);
-    io.to(user.room).emit("message", {
-      user: user.name,
-      text: message,
-      pfpSrc: user.pfpSrc,
+    socket.on("sendTyping", (typing, callback) => {
+        const user = getUser(socket.id);
+        console.log(user.name, typing);
+        io.to(user.room).emit("typing", {
+            user: user.name,
+            typing: typing,
+        });
     });
-    callback();
-  });
 
-  const leaveRoom = () => {
-    const user = removeUser(socket.id);
-    if (user) {
-      io.to(user.room).emit("message", {
-        user: "admin",
-        text: `${user.name} has left`,
-        roommates: getUsersInRoom(user.room),
-      });
-    }
-  };
+    const leaveRoom = () => {
+        const user = removeUser(socket.id);
+        if (user) {
+            io.to(user.room).emit("message", {
+                user: "admin",
+                text: `${user.name} has left`,
+                roommates: getUsersInRoom(user.room),
+            });
+        }
+    };
 
-  socket.on("leave", leaveRoom);
+    socket.on("leave", leaveRoom);
 
-  socket.on("disconnect", leaveRoom);
+    socket.on("disconnect", leaveRoom);
 });
 
 app.use(router);
 
 server.listen(PORT, () => {
-  console.log(`Server has started on port ${PORT}`);
+    console.log(`Server has started on port ${PORT}`);
```

Comment:
It is quite possible that we will have problems with code formatting, just as we have had in the past. 
For this we add a configuration with Prettier on the client side. However, I see that we did not include it on the server side.
Would you mind adding such configuration and files on the server side as well?

- [.prettierrc](https://github.com/manuelalferez/chatcus/blob/master/client/.prettierrc)
- [.prettierignore](https://github.com/manuelalferez/chatcus/blob/master/client/.prettierignore)